tests/std/tests/P0355R7_calendars_and_time_zones_clocks/test.cpp

```diff
@@ -0,0 +1,431 @@
+    assert(clock_cast<file_clock>(tt) == ft);
+    assert(clock_cast<file_clock>(gt) == ft);
+    assert(clock_cast<file_clock>(ft) == ft);
+
+    // [time.clock.utc.overview]/1 Example 1
+    assert(clock_cast<utc_clock>(sys_seconds{sys_days{1970y / January / 1}}).time_since_epoch() == 0s);
+    assert(clock_cast<utc_clock>(sys_seconds{sys_days{2000y / January / 1}}).time_since_epoch() == 946'684'822s);
+}
+
+static_assert(is_clock_v<utc_clock>);
+static_assert(is_clock_v<tai_clock>);
+static_assert(is_clock_v<gps_clock>);
+static_assert(is_clock_v<file_clock>);
+
+void test_utc_clock_from_sys(const leap_second& leap, seconds offset) {
+    // Generalized from [time.clock.utc.members]/3 Example 1.
+    auto t = leap.date() - 2ns;
+    auto u = utc_clock::from_sys(t);
+    assert(u.time_since_epoch() - t.time_since_epoch() == offset);
+
+    t += 1ns;
+    u = utc_clock::from_sys(t);
+    assert(u.time_since_epoch() - t.time_since_epoch() == offset);
+
+    t += 1ns;
+    u = utc_clock::from_sys(t);
+    offset += leap.value();
+    assert(u.time_since_epoch() - t.time_since_epoch() == offset);
+
+    t += 1ns;
+    u = utc_clock::from_sys(t);
+    assert(u.time_since_epoch() - t.time_since_epoch() == offset);
+}
+
+void test_file_clock_utc() {
+    const auto file_epoch{utc_clock::from_sys(sys_days{January / 1 / 1601})};
+    assert(file_clock::to_utc(file_time<seconds>{0s}) == file_epoch);
+    assert(file_clock::from_utc(file_epoch) == file_time<seconds>{0s});
+}
+
+void test_file_clock_to_utc(const leap_second& leap, seconds offset) {
+    // FILETIME counts leap seconds after 1 January 2017, so offset is constant thereafter.
+    constexpr sys_days file_time_cutoff{1d / January / 2017y};
+    auto t = clock_cast<file_clock>(leap.date()) - 2us;
+    if (leap > file_time_cutoff) {
+        offset = 27s;
+    }
+    offset -= duration_cast<seconds>(file_clock::duration{filesystem::__std_fs_file_time_epoch_adjustment});
+
+    auto u = file_clock::to_utc(t);
+    assert(u.time_since_epoch() - t.time_since_epoch() == offset);
+
+    t += 1us;
+    u = file_clock::to_utc(t);
+    assert(u.time_since_epoch() - t.time_since_epoch() == offset);
+
+    t += 1us;
+    u = file_clock::to_utc(t);
+    if (leap.date() <= file_time_cutoff)
+        offset += leap.value();
+    assert(u.time_since_epoch() - t.time_since_epoch() == offset);
+
+    t += 1us;
+    u = file_clock::to_utc(t);
+    assert(u.time_since_epoch() - t.time_since_epoch() == offset);
+}
+
+void test_gps_tai_clocks_utc() {
+    const auto tai_epoch{utc_clock::from_sys(sys_days{January / 1 / 1958} - seconds{10})};
+    const auto gps_epoch{utc_clock::from_sys(sys_days{January / Sunday[1] / 1980})};
+
+    assert(tai_clock::to_utc(tai_seconds{0s}) == tai_epoch);
+    assert(tai_clock::from_utc(tai_epoch) == tai_seconds{0s});
+
+    assert(gps_clock::to_utc(gps_seconds{0s}) == gps_epoch);
+    assert(gps_clock::from_utc(gps_epoch) == gps_seconds{0s});
+}
+
+int main() {

```

Comment:
I think we should also test each clock's `now()` function so that we at least exercise it somewhere!